Sources/SwiftKafka/KafkaProducerMessage.swift

```diff
@@ -0,0 +1,65 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the swift-kafka-gsoc open source project
+//
+// Copyright (c) 2022 Apple Inc. and the swift-kafka-gsoc project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of swift-kafka-gsoc project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Crdkafka
+import Foundation
+
+/// Message that is sent by the `KafkaProducer`
+public struct KafkaProducerMessage {
+    let topic: String
+    let partition: Int32
```

Comment:
Maybe it is useful to create a concrete type for partition that can be created with a `Int32` value and has a `static let` that is the `RD_KAFKA_PARTITION_UA` value under the hood.